src/euphorie/client/browser/report.py

```diff
@@ -252,4 +249,9 @@ def __call__(self):
             "Content-Type",
             "application/vnd.openxmlformats-" "officedocument.spreadsheetml.sheet",
         )
-        return save_virtual_workbook(book)
+
+        with NamedTemporaryFile(delete=False) as tmp:
+            save_workbook(book, tmp.name)
+            tmp.close()
+            with open(tmp.name, mode="rb") as f:
+                return f.read()
```

Comment:
tmp.seek(0)
```suggestion
            tmp.seek(0)
            return f.read()
```

Reply:
It feels right to me, because we're first writing and then reading. That's different modes.